src/main/java/fi/aalto/cs/apluscourses/ui/exercise/ExercisesView.java

```diff
@@ -67,12 +73,14 @@ public void viewModelChanged(@Nullable ExercisesTreeViewModel viewModel) {
         return;
       }
 
+      exerciseGroupsTree.removeMouseListener(mouseAdapter);
```

Comment:
Do we need to remove and re-add the listener always when the viewmodel changes? Can it just be there all the time?

Reply:
It can be there all the time